pkg/resource/deploy/plan_executor.go

```diff
@@ -184,6 +187,10 @@ func (pe *planExecutor) Execute(callerCtx context.Context, opts Options, preview
 			case <-ctx.Done():
 				logging.V(4).Infof("planExecutor.Execute(...): context finished: %v", ctx.Err())
 
+				if pe.stepGen.hasPolicyViolation {
```

Comment:
Let's move these out of this invocation and just do a single check below (near `pe.stepExec.Errored()`)